packages/mui-material/src/Autocomplete/Autocomplete.d.ts

```diff
@@ -339,29 +355,31 @@ export interface AutocompleteProps<
     ownerState: AutocompleteOwnerState<Value, Multiple, DisableClearable, FreeSolo, ChipComponent>,
   ) => React.ReactNode;
   /**
-   * Render the selected value.
+   * Render the selected value when doing multiple selections.
+   *
+   * @deprecated Use `renderValue` prop instead
    *
-   * @param {Value} value The `value` provided to the component.
-   * @param {function} tagProps The tag props.
+   * @param {Value[]} value The `value` provided to the component.
+   * @param {function} getTagProps A tag props getter.
    * @param {object} ownerState The state of the Autocomplete component.
    * @returns {ReactNode}
    */
-  renderSingleValue?: (
-    value: Value,
-    tagProps: AutocompleteTagProps,
+  renderTags?: (
+    value: Value[],
+    getTagProps: AutocompleteRenderGetTagProps,
     ownerState: AutocompleteOwnerState<Value, Multiple, DisableClearable, FreeSolo, ChipComponent>,
   ) => React.ReactNode;
   /**
-   * Render the selected value when doing multiple selections.
+   * Renders the selected value(s) as rich content in the input for both single and multiple selections.
    *
-   * @param {Value[]} value The `value` provided to the component.
+   * @param {AutocompleteRenderValue<Value, Multiple, FreeSolo>} value The `value` provided to the component.
    * @param {function} getTagProps A tag props getter.
    * @param {object} ownerState The state of the Autocomplete component.
    * @returns {ReactNode}
    */
-  renderTags?: (
-    value: Value[],
-    getTagProps: AutocompleteRenderGetTagProps,
+  renderValue?: (
+    value: AutocompleteRenderValue<Value, Multiple, FreeSolo>,
+    getTagProps: AutocompleteRenderValueGetTagProps<Multiple>,
```

Comment:
Since we're deprecating `renderTags`, wouldn't it make sense to deprecate the whole concept of tags and not mention the term `tag` in code? (@aarongarciah)
If so, this callback should be renamed.

Reply:
I had the same thought. The issue is that `getTagProps` returns properties specific to a Material UI Chip (or tag?), like `className` (`.MuiChip-root`), `disabled`, and `onDelete`. This callback is meant to be spread only when using a _custom_ Material UI Chip. Maybe we should rename it to `getChipProps`.

Reply:
I agree with replacing the term `tag` in our code. I think we can use `item` and explain in the docs what props are provided. We can explain in the docs that these include some props specific to Chip (`onDelete`) but other's that aren't. For example, the `data-tag-index` should be transformed to `data-item-index`, and this will be important to spread regardless of whether the component is a Chip or not.

Reply:
@michaldudak @DiegoAndai I've made the changes and updated the docs. I only replaced "tag" in the public API. Internally, some methods and variables still use "tag" since the deprecated `renderTags` depends on them. Renaming them would mean duplicating methods with the same logic for `renderValue`.